fast.py

```diff
@@ -1,58 +1,142 @@
+"""Load English-language names for FAST and LCSH subject classifications
+from N-Triple files acquired from data dumps.
+
+This is how we know, e.g. that FAST classification 1750175 means
+"Short stories, American".
+"""
+from contextlib import contextmanager
 import gzip
 import csv
+from io import BytesIO
 import logging
 import os
 import re
 from nose.tools import set_trace
 import time
+import zipfile
 
 class FASTNames(dict):
 
     SUBDIR = "FAST"
 
     triple_re = re.compile('^<http://id.worldcat.org/fast/([0-9]+)> <http://schema.org[#/]name> "([^"]+)"')
 
-    def load_filehandle(self, fh):
-        for triple in fh:
-            triple = triple.strip()
-            g = self.triple_re.search(triple)
-            if not g:
-                continue
-            identifier, name = g.groups()
-            self[identifier] = name
-
     @classmethod
     def from_data_directory(cls, data_directory):
+        """Load names from a directory that either contains a bunch of
```

Comment:
Is "names" the subject in each line of an N-Triple file? If I understand it correctly, if any row has an identifier but not a name, a name will be assigned?

Reply:
Yes, that's right. In our `subjects` table we have an "identifier" and a "name". The "identifier" is usually a number from a controlled vocabulary, like `sj00001253`. This script assigns the English "name" (e.g. "Ceratopsians") to any "identifiers" where we originally found out about the identifier but not the name.